test/extended/topology_manager/utils.go

```diff
@@ -292,26 +293,36 @@ func makeBusyboxPod(namespace string) *corev1.Pod {
 	}
 }
 
-func createPodsOnNodeSync(client clientset.Interface, namespace string, node *corev1.Node, testPods ...*corev1.Pod) []*corev1.Pod {
-	var updatedPods []*corev1.Pod
-	for _, testPod := range testPods {
-		if node != nil {
-			testPod.Spec.NodeSelector = map[string]string{
-				labelHostname: node.Name,
-			}
+func setNodeForPods(pods []*corev1.Pod, node *corev1.Node) {
+	for _, pod := range pods {
+		pod.Spec.NodeSelector = map[string]string{
+			labelHostname: node.Name,
 		}
+	}
 
-		created, err := client.CoreV1().Pods(namespace).Create(context.Background(), testPod, metav1.CreateOptions{})
-		e2e.ExpectNoError(err)
+}
 
-		err = waitForPhase(client, created.Namespace, created.Name, corev1.PodRunning, 5*time.Minute)
-		e2e.ExpectNoError(err)
+func createPods(client clientset.Interface, namespace string, testPods ...*corev1.Pod) []*corev1.Pod {
+	updatedPods := make([]*corev1.Pod, len(testPods), len(testPods))
+	var wg sync.WaitGroup
 
-		updatedPod, err := client.CoreV1().Pods(created.Namespace).Get(context.Background(), created.Name, metav1.GetOptions{})
-		e2e.ExpectNoError(err)
+	for i := 0; i < len(testPods); i++ {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+
+			created, err := client.CoreV1().Pods(namespace).Create(context.Background(), testPods[idx], metav1.CreateOptions{})
```

Comment:
Does this end up behaving the same as:
oc create -f pod1.yaml -f pod2.yaml, to make sure the issue in rhbz #1813567 is covered?

I believe that is the intent, but this code loop iterates over testPods, and it's not clear to me this will not end up being:
oc create -f pod1.yaml
oc create -f pod2.yaml

Running this test on openshift 4.4 should result in this test failing (as the fix is in 4.5, not 4.4). Can you please confirm?

Reply:
Yes, verification under 4.4 is currently in progress (I'm in the process of getting a cluster). So far I can say this pr causes no regressions. I'll update once done.

Reply:
solved! added explicit testcases, and verified it fails as expected on kube 1.17.1.